File: lc_soln/0217_contains_duplicate/solution.py
from typing import List


class Solution:
    def has_duplicate(self, nums: List[int]) -> bool:
        seen: set[int] = set()
        for num in nums:
            if num in nums:
                return True
            seen.add(num)
        return False


def main():
    print(Solution().has_duplicate(nums=[1, 2, 3, 3]))


if __name__ == "__main__":
    main()

File: lc_soln/0217_contains_duplicate/solution.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        var seen = new HashSet<int>();
        foreach (int num in nums) {
            if (seen.Contains(num)) {
                return true;
            }
            seen.Add(num);
        }
        return false;
    }
}

class Program {
    static void Main() {
        var solution = new Solution();
        Console.WriteLine(solution.hasDuplicate([1, 2, 3, 3]));
    }
}
